easybuild/tools/modules.py

```diff
@@ -115,9 +115,31 @@
 _log = fancylogger.getLogger('modules', fname=False)
 
 
+MODULE_AVAIL_CACHE = {}
+MODULE_SHOW_CACHE = {}
 MODULE_VERSION_CACHE = {}
 
 
+def reset_module_caches():
+    """Reset module caches."""
+    MODULE_AVAIL_CACHE.clear()
+    MODULE_SHOW_CACHE.clear()
+
+
+def invalidate_module_caches_for(path):
+    """Invalidate cache entries related to specified path."""
+    _log.debug("Invallidating module cache entries for path '%s'", path)
+    for cache, subcmd in [(MODULE_AVAIL_CACHE, 'avail'), (MODULE_SHOW_CACHE, 'show')]:
+        for key in cache.keys():
+            paths_in_key = '='.join(key[0].split('=')[1:]).split(os.pathsep)
+            _log.debug("Paths for 'module %s' key '%s': %s", subcmd, key, paths_in_key)
+            for path_in_key in paths_in_key:
+                if os.path.samefile(path, path_in_key):
```

Comment:
hmm, may need to be more careful here, `samefile` throws an error if either of these paths don't exist